drivers/wdm/audio/hdaudbus/fdo.cpp

```diff
@@ -120,7 +120,7 @@ HDA_DpcForIsr(
         if (Codec->ResponseCount >= MAX_CODEC_RESPONSES)
         {
             DPRINT1("too many responses for codec %x Response %x ResponseFlags %x\n", Cad, Response, ResponseFlags);
-            continue;
+            break;
```

Comment:
This is not a device enumeration code, but a reading RIRB code. If one item is wrong here, we shouldn't stop reading it as other entries may be correct.
If this code loops for you, the reason is definitely something else.

> Don't dumbly repeat Haiku OS behaviour.

Are you sure Haiku does the same here? :)

Reply:
> This is not a device enumeration code, but a reading RIRB code. If one item is wrong here, we shouldn't stop reading it as other entries may be correct.
If this code loops for you, the reason is definitely something else.

Ah, okay then. As I've suspected. I think the problem is too small max count of RIRB items defined by Haiku's driver.h:
https://github.com/reactos/reactos/blob/master/drivers/wdm/audio/hdaudbus/driver.h#L43
Intel HD Audio specification declares 3 possible counts for RIRB responses:
![RIRB-size](https://user-images.githubusercontent.com/26385117/211217966-cd62ed5b-5634-40e3-bf88-0c77a92bf839.png)
https://www.intel.com/content/dam/www/public/us/en/documents/product-specifications/high-definition-audio-specification.pdf
2, 16 and 256.
Haiku's one is limited to 16, but I think it's not enough. Realtek HD Audio seems require 256, and in general, the maximum possible amount of items should be 256, to succeed for all types of the controllers. **Indeed, with 256, the installation doesn't fail anymore, although it installs 3 minutes in debug mode, and the count overflow case isn't reached.** :smiley: So yes, this is not a correct fix, so I should remove it, but set `MAX_CODEC_RESMPONSES` to 256.
> Are you sure Haiku does the same here? :)

Yep, just look at this code:
https://github.com/haiku/haiku/blob/ee4577937404e1f320bfde245f4dea1739735a92/src/add-ons/kernel/drivers/audio/hda/hda_controller.cpp#L401
It's obviously looks similar. :smile:

Reply:
You are mixing `MAX_CODEC_RESPONSES` and RIRB size. RIRB size is represented by `DeviceExtension->RirbLength` variable, and is indeed set into one of 3 values inside `HDA_InitCorbRirbPos`.
`MAX_CODEC_RESPONSES` is a queue size where responses are stored for a specific codec. The fact that the `Codec->ResponseCount >= MAX_CODEC_RESPONSES` condition hits here, mean that the queue is full and nobody takes elements from it.